js/songs/form.directive.js

```diff
@@ -0,0 +1,30 @@
+(function(){
+  "use strict";
+  angular
+    .module("songs")
+    .directive("songForm", [
+      "SongFactory",
+      "$state",
+      FormDirectiveFunction
+    ]);
+
+  function FormDirectiveFunction(SongFactory, $state){
+    return{
+      templateUrl: "js/songs/_form.html",
+      restrict: "E",
+      scope: {
+        song: "="
+      },
+      link: function(scope){
+          scope.create = function(scope){
+            console.log("create?");
+            scope.song.$save(function(response){
+              console.log("create is called?");
+              $state.go("songs", {}, {reload: true});
```

Comment:
i like this debugging here, trying to figure out where things our happening. What was your findings? If only `create?` shows up then you know the functions firing, but probably song is not an angular factory object.